ios/Exponent/Versioned/Core/Api/EXRandomBytes.m

```diff
@@ -0,0 +1,64 @@
+//
+//  EXRandomBytes.m
+//  Exponent
+//
+//  Created by Evan Bacon on 10/24/18.
+//  Copyright © 2018 650 Industries. All rights reserved.
+//
+
+/*
+  The MIT License (MIT)
+
+  Copyright (c) 2015 Mark Vayngrib
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+*/
+
+#import "EXRandomBytes.h"
+#import <React/RCTBridgeModule.h>
+
+@implementation EXRandomBytes
+
+RCT_EXPORT_MODULE(RNRandomBytes)
```

Comment:
```suggestion
RCT_EXPORT_MODULE(ExpoRandom)
```

Reply:
> ExpoRandom
I think the core purpose of this library is to emulate the node.js `crypto` function:
```js
// Asynchronous
const crypto = require('crypto');
crypto.randomBytes(256, (err, buf) => {
  if (err) throw err;
  console.log(`${buf.length} bytes of random data: ${buf.toString('hex')}`);
});
```
Syntax wise, you shim the library and use it the same way. 
In order to make this useful with crypto, one must implement [`rn-nodeify`](https://github.com/tradle/rn-nodeify) which is a pretty big hack. 

But say we rename the `NativeModule`, we would need to fork both: 
* https://github.com/tradle/react-native-crypto/blob/master/index.js
* https://github.com/mvayngrib/react-native-randombytes/blob/master/index.js

IMO this isn't a lot of work, but if we are going to go that far we should just rewrite everything. 

Neither library is very popular or widely adopted, but this would also prioritize lower than other modules.

Reply:
Node's `crypto.randomBytes` has a synchronous option (by omitting the callback) and also returns a Buffer (TypedArray would likely work too) instead of a bas64-encoded string. So regardless we'd need a custom wrapper around the native API we expose in order to implement something that resemble's the Node API.

We also need to consider https://developer.mozilla.org/en-US/docs/Web/API/Crypto (both targeting it and making it possible for someone to write a polyfill for it using Expo's APIs as the underlying implementation).